optimade/filtertransformers/base_transformer.py

```diff
@@ -22,29 +67,101 @@ class BaseTransformer(abc.ABC, Transformer):
         "=": None,
     }
 
-    # map from back-end specific operators to their inverse
-    # e.g. {"$lt": "$gt"} for MongoDB.
-    reversed_operator_map: Dict[str, str] = {}
+    # map from operators to their syntactic (as opposed to logical) inverse to handle
+    # equivalence between cases like "A > 3" and "3 < A".
+    _reversed_operator_map = {
+        ">": "<",
+        ">=": "<=",
+        "<": ">",
+        "<=": ">=",
+        "=": "=",
+        "!=": "!=",
+    }
+
+    __quantity_type: Type[Quantity] = Quantity
+    __quantities = None
```

Comment:
Hmmm... I think there is a semantic (if fuzzy) distinction between mangled private vars/methods and dunder methods. On a superficial level, our docs treat them differently so these would both appear in the rendered docs (which we don't want). 

- The use here for `__quantities` is just a hack to make a cached property (`@functools.cached_property` was only added in 3.8)
- The "privitisation" of `quantity_type` here is also primarily docs-related... if it is public/protected, then mkdocs/mkdocstrings/pytkdocs/whatever is responsible ends up inserting *the whole* definition of the underlying `Type`, rather than just the fact that we are expecting a type. I consider this a bug...